MorkoUnityProject/Assets/Code/LocalPlayerController.cs

```diff
@@ -245,109 +244,69 @@ private void Move(Vector3 moveDirection, bool accelerateRun)
-				rightJoystickForRotation = false;
+				targetRotation = mouseRotation;
 			}
 			else if (mouseAngle < joystickAngle)
 			{
-				mouseForRotation = false;
 				mouseRotatedLast = false;
-				rightJoystickForRotation = true;
+				targetRotation = Quaternion.LookRotation(lookDirectionJoystick, Vector3.up);
 			}
 		}
-		else if (hasMoved)
-			onlyLeftJoystickUsed = true;
-		
+		else if (hasMoved && currentSettings.rotateTowardsMove)
+			targetRotation = Quaternion.LookRotation(moveDirection);
 
-		if (onlyLeftJoystickUsed && !mouseRotatedLast && currentSettings.rotateTowardsMove)
-		{
-			character.transform.rotation = Quaternion.LookRotation(moveDirection);
-		}
-		else if (mouseForRotation)
-		{
-			lastMousePosition = currentMousePosition;
-			Ray mouseRay = camera.ScreenPointToRay(Input.mousePosition);
-			RaycastHit hit;
-			Physics.Raycast(mouseRay, out hit, groundMask);
-			Vector3 lookDirection = (hit.point - character.transform.position).normalized;
-			Vector3 lookDirectionLevel = new Vector3(lookDirection.x, character.transform.position.y, lookDirection.z);
-			character.transform.rotation = Quaternion.LookRotation(lookDirectionLevel);
-		}
-		else if (rightJoystickForRotation)
-		{
-			Quaternion lookRotation = Quaternion.LookRotation(lookDirectionJoystick, Vector3.up);
-			character.transform.rotation = Quaternion.RotateTowards(lookRotation, character.transform.rotation, Time.deltaTime);
-		}
-		
-		// If rotation amount > threshold, slowdown character
-		float angle = Vector3.Angle(character.transform.forward, lastRotation);
-		lastRotation = character.transform.forward;
+		character.transform.rotation = targetRotation;
 	}
-	
-	private void Dash(bool dive)
-	{
-		if (!isMorko || !dive) return;
 
-		Vector3 currentPosition = character.transform.position;
-		Vector3 targetPosition = currentPosition + character.transform.forward * dashDistance;
-		
-		long currentMillis = DateTime.Now.Ticks / TimeSpan.TicksPerMillisecond;
-		bool pastCooldown = currentMillis - lastMillis >= dashCooldown * 1000;
-			
-		if (!pastCooldown)
-			return;
-		
-		var time = 0f;
-		while(time < 1)
+	private Quaternion GetMouseRotation()
+	{
+		Plane playerPlane = new Plane(Vector3.up, character.transform.position);
+		Ray ray = camera.ScreenPointToRay (Input.mousePosition);
+         
+		float distance = 0.0f;
+		var mouseRotation = character.transform.rotation;
+         
+		if (playerPlane.Raycast (ray, out distance)) 
 		{
-			time += Time.deltaTime / dashDuration;
-			Debug.Log(time);
-			character.transform.position = Vector3.Lerp(currentPosition, targetPosition, time);
-			
-			// If collision with other player
-			// changeState(false);
-			// Change other player to morko
+			Vector3 targetPoint = ray.GetPoint(distance);
+			mouseRotation = Quaternion.LookRotation(targetPoint - character.transform.position);
 		}
-		
-		lastMillis = DateTime.Now.Ticks / TimeSpan.TicksPerMillisecond;
-		
-		currentMovementSpeed = 0f;
+
+		return mouseRotation;
 	}
```

Comment:
funktion nimi ei liity mitenkään siihen mitä siinä tehdään:D 
(in english by google translate: the function name has nothing to do with what it does: D)
pitäis nykysellään olla RotationToCursorPositionRelativeToCameraPositionAndPlayerPosition() jos haluis et se vastais sitä mitä se tekee